service/src/lib.rs

```diff
@@ -289,7 +293,7 @@ pub trait Service: Send + Sync {
     /// This function is run exactly once on each instance of a deployment.
     ///
     /// The deployer expects this instance of [Service][Service] to bind to the passed [SocketAddr][SocketAddr].
-    fn bind(&mut self, addr: SocketAddr) -> Result<(), error::Error>;
+    fn bind(&mut self, addr: SocketAddr) -> Result<ServeHandle, error::Error>;
```

Comment:
Probably need to bump the version since the API is breaking.

Reply:
I can, but I'm breaking it again in the PR that follows this one. Should we rather only bump once?

Reply:
Fair enough - I'm not sure to be honest but we need a consistent strategy. Tbf we have been consistent in bumping with distinct commits for that purpose, so let's stick to that for now I guess :)